NExtends/Html/HtmlUtility.cs

```diff
@@ -0,0 +1,213 @@
+
+            HtmlDocument html = GetHtml(source);
+            if (html == null) { return string.Empty; }
+
+            HtmlNode allNodes = html.DocumentNode;
+            CleanNodes(allNodes, new HashSet<string>(ValidHtmlTags.Keys));
+
+            foreach (KeyValuePair<string, string[]> tag in ValidHtmlTags)
+            {
+                IEnumerable<HtmlNode> nodes = allNodes.DescendantsAndSelf().Where(n => n.Name == tag.Key);
+
+                // No nodes? Skip.
+                if (nodes == null) { continue; }
+
+                foreach (HtmlNode node in nodes)
+                {
+                    if (!node.HasAttributes) { continue; }
+
+                    foreach (HtmlAttribute a in node.Attributes)
+                    {
+                        if (!tag.Value.Contains(a.Name))
+                        {
+                            a.Remove();
+                        }
+                        else if (a.Name == "href" || a.Name == "src")
+                        {
+                            a.Value = (!string.IsNullOrEmpty(a.Value)) ? a.Value.Replace("\r", "").Replace("\n", "") : "";
+                            a.Value = (!string.IsNullOrEmpty(a.Value) && (a.Value.IndexOf("javascript") < 10 || a.Value.IndexOf("eval") < 10))
+                                    ? a.Value.Replace("javascript", "").Replace("eval", "")
+                                    : a.Value;
+                        }
+                        else if (a.Name == "class" || a.Name == "style")
+                        {
+                            a.Value = Microsoft.Security.Application.Encoder.CssEncode(a.Value);
+                        }
+                        else
+                        {
+                            a.Value = Microsoft.Security.Application.Encoder.HtmlAttributeEncode(a.Value);
+                        }
+                    }
+                }
+            }
+
+            return allNodes.InnerHtml;
+        }
+
+        /// <summary>
+        /// Takes a raw source and removes all HTML tags
+        /// </summary>
+        /// <param name="source"></param>
+        /// <returns></returns>
+        public string StripHtml(string source)
+        {
+            if (string.IsNullOrEmpty(source))
+            {
+                return string.Empty;
+            }
+
+            source = SanitizeHtml(source);
+
+            HtmlDocument html = GetHtml(source);
+            var result = new StringBuilder();
+
+            foreach (HtmlNode node in html.DocumentNode.ChildNodes)
+            {
+                result.Append(node.InnerText);
+            }
+
+            return result.ToString();
+        }
+
+        /// <summary>
+        /// Recursively delete nodes not in the whitelist
+        /// </summary>
+        private static void CleanNodes(HtmlNode node, HashSet<string> whitelist)
+        {
+            if (node.NodeType == HtmlNodeType.Element)
+            {
+                if (!whitelist.Contains(node.Name))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1066)